Android: cannot find -lpthread for simple binary

I'm trying to compile a simple application for Android and getting linker errors: Bazel is linking in `-lpthread`, even though Android doesn't support that.

The application code is as follows:

*tensorflow/demo-bug/main.cc*:
```
int main() {
  return 0;
}
```

The BUILD file is as follows:

*tensorflow/demo-bug/BUILD*:
```
cc_binary(
    name = "main",
    srcs = ["main.cc"],
    deps = ["//tensorflow/core:core_cpu"],
)
```

If you don't include `//tensorflow/core:core_cpu` in the `deps`, the error doesn't happen. For some reason, that dependency pulls in the `-lpthread` from somewhere.

When you run the `bazel build` command (see below), you get the following error:
```
external/androidndk/ndk/toolchains/arm-linux-androideabi-4.9/prebuilt/linux-x86_64/lib/gcc/arm-linux-androideabi/4.9.x/../../../../arm-linux-androideabi/bin/ld: error: cannot find -lpthread
```
Indeed, looking through the command that it uses to compile it, you see the flag:
```
...-stl/llvm-libc++/libs/armeabi-v7a/libunwind.a -ldl -lm -ldl -lpthread -lm -pthread -lm -lm -static-libgcc -gcc-toolchain external/androidndk/ndk/toolchains/arm-linux-androideabi-4.9/prebuilt/linux-x86_64 -no-canonical-prefixes ...
```
(Full command below)
```
  external/androidndk/ndk/toolchains/llvm/prebuilt/linux-x86_64/bin/clang -o bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/demo-bug/main bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/demo-bug/_objs/main/tensorflow/demo-bug/main.o -Wl,-whole-archive bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/libcore_cpu_internal.lo -Wl,-no-whole-archive bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/grappler/optimizers/libmeta_optimizer.a bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/grappler/optimizers/libarithmetic_optimizer.a bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/grappler/optimizers/libgraph_optimizer_stage.a bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/grappler/optimizers/libauto_parallel.a bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/grappler/optimizers/libcustom_graph_optimizer_registry_impl.a bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/grappler/optimizers/libdebug_stripper.a bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/grappler/optimizers/libdependency_optimizer.a bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/grappler/optimizers/libfunction_optimizer.a bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/grappler/optimizers/liblayout_optimizer.a bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/grappler/optimizers/libloop_optimizer.a bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/grappler/optimizers/libmemory_optimizer.a bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/grappler/optimizers/libstatic_schedule.a bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/grappler/costs/libgraph_memory.a bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/grappler/clusters/libvirtual_cluster.a bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/grappler/costs/libop_level_cost_estimator.a bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/grappler/costs/libvirtual_scheduler.a bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/grappler/costs/libvirtual_placer.a bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/grappler/libdevices.a bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/grappler/utils/libtraversal.a bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/grappler/optimizers/libmodel_pruner.a bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/grappler/optimizers/libgraph_rewriter.a bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/grappler/optimizers/libremapper.a bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/grappler/optimizers/libconstant_folding.a bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/grappler/optimizers/libscoped_allocator_optimizer.a -Wl,-whole-archive bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/libscoped_allocator_ops_op_lib.lo -Wl,-no-whole-archive bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/grappler/optimizers/libshape_optimizer.a bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/grappler/optimizers/libsymbolic_shapes.a bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/grappler/costs/libgraph_properties.a bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/grappler/costs/libutils.a bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/grappler/clusters/libutils.a bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/libgpu_id_impl.a bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/grappler/libgraph_view.a bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/grappler/clusters/libcluster.a bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/grappler/utils/libframe.a -Wl,-whole-archive bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/libcore_cpu_base.lo -Wl,-no-whole-archive -Wl,-whole-archive bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/libfunction_ops_op_lib.lo -Wl,-no-whole-archive -Wl,-whole-archive bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/libfunctional_grad.lo -Wl,-no-whole-archive -Wl,-whole-archive bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/libfunctional_ops_op_lib.lo -Wl,-no-whole-archive bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/grappler/utils/libcolocation.a bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/grappler/utils/libfunctions.a bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/grappler/utils/libtopological_sort.a -Wl,-whole-archive bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/kernels/libno_op.lo -Wl,-no-whole-archive -Wl,-whole-archive bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/kernels/libsendrecv_ops.lo -Wl,-no-whole-archive -Wl,-whole-archive bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/libno_op_op_lib.lo -Wl,-no-whole-archive -Wl,-whole-archive bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/libsendrecv_ops_op_lib.lo -Wl,-no-whole-archive -Wl,-whole-archive bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/libcore_cpu_impl.lo -Wl,-no-whole-archive bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/libgraph.a bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/grappler/libgrappler_item.a bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/grappler/libop_types.a bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/grappler/libutils.a -Wl,-whole-archive bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/libframework_internal_impl.lo -Wl,-no-whole-archive bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/libprotos_all_proto_text.a bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/liberror_codes_proto_text.a bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/liblib_internal_impl.a bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/liblib_hash_crc32c_accelerate_internal.a bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/liblib_proto_parsing.a bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/libabi.a bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/libcore_stringpiece.a bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/libplatform_base.a bazel-out/armeabi-v7a-py3-opt/bin/external/snappy/libsnappy.a bazel-out/armeabi-v7a-py3-opt/bin/external/double_conversion/libdouble-conversion.a bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/grappler/costs/libop_performance_data_cc_impl.a bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/libversion_lib.a bazel-out/armeabi-v7a-py3-opt/bin/external/gif_archive/libgif.a bazel-out/armeabi-v7a-py3-opt/bin/external/jpeg/libjpeg.a bazel-out/armeabi-v7a-py3-opt/bin/external/jpeg/libsimd_armv7a.a bazel-out/armeabi-v7a-py3-opt/bin/external/com_googlesource_code_re2/libre2.a bazel-out/armeabi-v7a-py3-opt/bin/external/farmhash_archive/libfarmhash.a bazel-out/armeabi-v7a-py3-opt/bin/external/fft2d/libfft2d.a bazel-out/armeabi-v7a-py3-opt/bin/external/highwayhash/libsip_hash.a bazel-out/armeabi-v7a-py3-opt/bin/external/highwayhash/libarch_specific.a bazel-out/armeabi-v7a-py3-opt/bin/external/png_archive/libpng.a bazel-out/armeabi-v7a-py3-opt/bin/external/zlib_archive/libzlib.a bazel-out/armeabi-v7a-py3-opt/bin/external/nsync/libnsync_cpp.a bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/libprotos_all_proto_cc_impl.a bazel-out/armeabi-v7a-py3-opt/bin/tensorflow/core/liberror_codes_proto_cc_impl.a bazel-out/armeabi-v7a-py3-opt/bin/external/protobuf_archive/libprotobuf.a bazel-out/armeabi-v7a-py3-opt/bin/external/protobuf_archive/libprotobuf_lite.a external/androidndk/ndk/sources/cxx-stl/llvm-libc++/libs/armeabi-v7a/libandroid_support.a external/androidndk/ndk/sources/cxx-stl/llvm-libc++/libs/armeabi-v7a/libc++.a external/androidndk/ndk/sources/cxx-stl/llvm-libc++/libs/armeabi-v7a/libc++_static.a external/androidndk/ndk/sources/cxx-stl/llvm-libc++/libs/armeabi-v7a/libc++abi.a external/androidndk/ndk/sources/cxx-stl/llvm-libc++/libs/armeabi-v7a/libunwind.a -ldl -lm -ldl -lpthread -lm -pthread -lm -lm -static-libgcc -gcc-toolchain external/androidndk/ndk/toolchains/arm-linux-androideabi-4.9/prebuilt/linux-x86_64 -no-canonical-prefixes -target armv7-none-linux-androideabi -Wl,--fix-cortex-a8 -Lexternal/androidndk/ndk/sources/cxx-stl/llvm-libc++/libs/armeabi-v7a '--sysroot=external/androidndk/ndk/platforms/android-23/arch-arm')
```

### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Yes, see above
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 16.04
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**: N/A
- **TensorFlow installed from (source or binary)**: Source
- **TensorFlow version (use command below)**: v1.9.0 (git tag)
- **Python version**: 3.5
- **Bazel version (if compiling from source)**: 0.15.2
- **GCC/Compiler version (if compiling from source)**: Android NDK r17b
- **CUDA/cuDNN version**: N/A
- **GPU model and memory**: N/A
- **Exact command to reproduce**:
```
# ARM v7
bazel \
  --crosstool_top=//external:android/crosstool \
  --host_crosstool_top=@bazel_tools//tools/cpp:toolchain \
  --config=android \
  --cpu=armeabi-v7a \
  --fat_apk_cpu=armeabi-v7a \
  --verbose_failures \
  --cxxopt=-std=c++11 \
  --config=monolithic \
  //tensorflow/demo-bug:main

# ARM v8
bazel \
  --crosstool_top=//external:android/crosstool \
  --host_crosstool_top=@bazel_tools//tools/cpp:toolchain \
  --config=android \
  --cpu=arm64-v8a \
  --fat_apk_cpu=arm64-v8a \
  --verbose_failures \
  --cxxopt=-std=c++11 \
  --config=monolithic \
  //tensorflow/demo-bug:main
```

--- a/tensorflow/core/BUILD
+++ b/tensorflow/core/BUILD
@@ -2037,6 +2037,7 @@ cc_library(
         "//tensorflow:freebsd": [],
         "//tensorflow:windows": [],
         "//tensorflow:windows_msvc": [],
+        "//tensorflow:android": [],
         "//conditions:default": [
             "-ldl",
             "-lpthread",

